conans/client/conan_api.py

```diff
@@ -535,6 +541,45 @@ def raise_if_not_exists(some_path):
                 raise_if_not_exists(conanfile_path)
         return conanfile_path
 
+    @staticmethod
+    def _get_conanfile_path_from_dir(path):
+        """
+        If path is a directory, apends 'conanfile.py' and returns its path. If not,
+        it checks if path parameter is file and ends with '.py', otherwise it raises an exception
+
+        :param path: Path to a conanfile.py or directory containing it
+        :return: Conanfile path checked as file ending with '.py'
+        '"""
+        if os.path.isdir(path):
+            path = os.path.join(path, CONANFILE)
+
+        if not os.path.isfile(path):
+            raise ConanException("Conanfile not found: %s" % path)
+
+        if not path.endswith(".py"):
+            raise ConanException("A conanfile.py is needed (not valid conanfile.txt)")
+
+        return path
+
+    @staticmethod
+    def _get_dir_from_conanfile_path(path):
+        """
```

Comment:
Maybe too many checks, you already checked in the `_get_conanfile_path_from_dir`.
Could if be just `return os.path.dirname(path) if os.path.isfile(path) else path`?  Maybe an `os.path.exists`?

Reply:
I think both is better: `os.path.isfile(path)` and `os.path.exists(path)` just to make sure it is an existing path